From 6c47f7c4435e498e9c4707e80f869276749c5a21 Mon Sep 17 00:00:00 2001
Message-Id: <6c47f7c4435e498e9c4707e80f869276749c5a21.1440106296.git.chang-joon.lee@intel.com>
In-Reply-To: <1f426693c14c79e65ad8e62ef6bb4b07d6484040.1440106296.git.chang-joon.lee@intel.com>
References: <1f426693c14c79e65ad8e62ef6bb4b07d6484040.1440106296.git.chang-joon.lee@intel.com>
From: Dhanya <dhanya.p.r@intel.com>
Date: Thu, 2 Jul 2015 10:18:16 +0530
Subject: [PATCH 04/11] FOR_UPSTREAM [VPG]: drm/i915: Remove redundant
 PORT_HOTPLUG_STAT reads

For DP hotplug,there is a redundant  check for PORT_HOTPLUG_STAT in irq_handler
and in g4x_dp_detect(intel_dp.c),which is causing issue in enabling display
less validation for DP.Hence removing the check in intel_dp.c.

Also removed the switch case in g4x_dp_detect() as its a dead code now.

Change-Id: I8d803b8d84b940d819dacd8ed20bfbc08e5ebacf
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-14778
Signed-off-by: dhanya.p.r@intel.com

## Don't forget to add a JIRA URL to the Tracked-On metadata!
## https://wiki.ith.intel.com/display/CACTUS/Patch+requirements+check
## url should be in the form:
## Tracked-On: https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
## Several Tracked-On lines are allowed
## you can fully prefill your Tracked-On commits using environment variable:
## export CURRENT_TRACKED_ON=https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
##
---
 drivers/gpu/drm/i915/intel_dp.c | 36 ------------------------------------
 1 file changed, 36 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 7a062f3..dc1d198 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -4474,9 +4474,6 @@ static enum drm_connector_status
 g4x_dp_detect(struct intel_dp *intel_dp)
 {
 	struct drm_device *dev = intel_dp_to_dev(intel_dp);
-	struct drm_i915_private *dev_priv = dev->dev_private;
-	struct intel_digital_port *intel_dig_port = dp_to_dig_port(intel_dp);
-	uint32_t bit;
 
 	/* Can't disconnect eDP, but you can close the lid... */
 	if (is_edp(intel_dp)) {
@@ -4488,39 +4485,6 @@ g4x_dp_detect(struct intel_dp *intel_dp)
 		return status;
 	}
 
-	if (IS_VALLEYVIEW(dev)) {
-		switch (intel_dig_port->port) {
-		case PORT_B:
-			bit = PORTB_HOTPLUG_LIVE_STATUS_VLV;
-			break;
-		case PORT_C:
-			bit = PORTC_HOTPLUG_LIVE_STATUS_VLV;
-			break;
-		case PORT_D:
-			bit = PORTD_HOTPLUG_LIVE_STATUS_VLV;
-			break;
-		default:
-			return connector_status_unknown;
-		}
-	} else {
-		switch (intel_dig_port->port) {
-		case PORT_B:
-			bit = PORTB_HOTPLUG_LIVE_STATUS_G4X;
-			break;
-		case PORT_C:
-			bit = PORTC_HOTPLUG_LIVE_STATUS_G4X;
-			break;
-		case PORT_D:
-			bit = PORTD_HOTPLUG_LIVE_STATUS_G4X;
-			break;
-		default:
-			return connector_status_unknown;
-		}
-	}
-
-	if ((I915_READ(PORT_HOTPLUG_STAT) & bit) == 0)
-		return connector_status_disconnected;
-
 	/* Avoid DPCD opertations if status is same */
 	if (intel_dp->attached_connector->base.status ==
 				connector_status_connected)
-- 
1.9.1

